tests/integration/security/pass_through_filter_chain_test.go

```diff
@@ -225,9 +225,9 @@ func TestPassThroughFilterChain(t *testing.T) {
 				}
 				ctx.NewSubTest(fmt.Sprintf("In %s", cluster.Name())).Run(func(ctx framework.TestContext) {
 					for _, tc := range cases {
-						name := fmt.Sprintf("G->%s:%d[%t]", tc.target.Config().Service, tc.port, tc.want)
-						g := apps.G.Match(echo.InCluster(cluster)).GetOrFail(ctx, echo.Namespace(ns.Name()))
-						from := getWorkload(g, t)
+						name := fmt.Sprintf("A->%s:%d[%t]", tc.target.Config().Service, tc.port, tc.want)
+						a := apps.A.Match(echo.InCluster(cluster)).GetOrFail(ctx, echo.Namespace(ns.Name()))
+						from := getWorkload(a, t)
```

Comment:
will this change the semantic of the test for a->a? previous the source pod is always different from the target, and now it's from the same pod.

Reply:
Hi Yangmin, thanks for the reviewing! 
> A general question, is the refactor just to reduce the number of echo deployments for performance optimization?
How much more do you plan to reduce and do we know how much it improves the performance?

Yes, this is for https://github.com/istio/istio/issues/29475. Actually currently with the number of clusters reduced to 3, the number of deployed echos is reduced to 47 and job time is around 25 mins. This PR will reduce the number of echos to 41. The improvement is not much. @howardjohn Do you think the current 47 is OK? If it is OK, we can keep the G and X so that the test logic is more straightforward.